beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestWithInArrearGreedySubscriptions.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright 2020-2022 Equinix, Inc
+ * Copyright 2014-2022 The Billing Project, LLC
+ *
+ * The Billing Project licenses this file to you under the Apache License, version 2.0
+ * (the "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at:
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.killbill.billing.beatrix.integration;
+
+import java.math.BigDecimal;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.joda.time.LocalDate;
+import org.killbill.billing.account.api.Account;
+import org.killbill.billing.api.TestApiListener.NextEvent;
+import org.killbill.billing.beatrix.util.InvoiceChecker.ExpectedInvoiceItemCheck;
+import org.killbill.billing.catalog.api.BillingActionPolicy;
+import org.killbill.billing.catalog.api.PlanPhaseSpecifier;
+import org.killbill.billing.entitlement.api.DefaultEntitlement;
+import org.killbill.billing.entitlement.api.DefaultEntitlementSpecifier;
+import org.killbill.billing.entitlement.api.Entitlement;
+import org.killbill.billing.entitlement.api.Entitlement.EntitlementActionPolicy;
+import org.killbill.billing.entitlement.api.Entitlement.EntitlementState;
+import org.killbill.billing.invoice.api.InvoiceItemType;
+import org.killbill.billing.payment.api.PluginProperty;
+import org.killbill.billing.platform.api.KillbillConfigSource;
+import org.killbill.billing.util.config.definition.InvoiceConfig.InArrearMode;
+import org.killbill.billing.util.config.definition.InvoiceConfig.UsageDetailMode;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import com.google.common.collect.ImmutableList;
```

Comment:
No guava on master 😄

Reply:
Indeed! Fix in https://github.com/killbill/killbill/pull/1827/commits/152903d48327e4ba4a381605d65b13fb6b1a1c62

@xsalefter I am wondering if we could find a way to make the build fail if/when we pull guava classes. There is no explicit dependency in the `pom.xml` - I suppose it must be working as it is a transitive dependency from `guice` ?

Reply:
Yes, seems like it is transitive dependency from guice. 

> I am wondering if we could find a way to make the build fail if/when we pull guava classes.

I tried to simulate this locally (create class that contains `ImmutableList`) and get:
```
[ERROR] Used undeclared dependencies found:
[ERROR]    com.google.guava:guava:jar:31.1-jre:provided
```

Now why CI passed this? :thinking:

Reply:
Maybe because [skipped here](https://github.com/killbill/gh-actions-shared/blob/main/.github/workflows/ci.yml#L15)?

Reply:
CI **did** break (the `verify` job, not the test jobs). The reason I run these checks separately (in their own `verify` job) is because they take a long time (best to do it once vs doing it for every test suite).

<img width="407" alt="image" src="https://user-images.githubusercontent.com/51940/211144605-75708570-e5d9-43fc-8f84-adaf251d9074.png">